exporter/datadogexporter/logs_exporter.go

```diff
@@ -0,0 +1,104 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter"
+
+import (
+	"context"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/source"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/logs"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/scrub"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/utils"
+)
+
+type logsExporter struct {
+	params         component.ExporterCreateSettings
+	cfg            *Config
+	ctx            context.Context // ctx triggers shutdown upon cancellation
+	scrubber       scrub.Scrubber  // scrubber scrubs sensitive information from error messages
+	sender         *logs.Sender
+	onceMetadata   *sync.Once
+	sourceProvider source.Provider
+}
+
+// NewExporter creates a new instance of LogsExporter
+func newLogsExporter(ctx context.Context, params component.ExporterCreateSettings, cfg *Config, onceMetadata *sync.Once, sourceProvider source.Provider) (*logsExporter, error) {
+	// create datadog client
+	// validation endpoint is provided by Metrics
+	client := utils.CreateClient(cfg.API.Key, cfg.Metrics.TCPAddr.Endpoint)
+	// validate the apiKey
+	if err := utils.ValidateAPIKey(params.Logger, client); err != nil && cfg.API.FailOnInvalidKey {
+		return nil, err
+	}
+
+	s := logs.NewSender(cfg.Logs.TCPAddr.Endpoint, params.Logger, cfg.TimeoutSettings, cfg.LimitedHTTPClientSettings.TLSSetting.InsecureSkipVerify, cfg.API.Key)
```

Comment:
Might be nicer to pass a struct `SenderConfig` or something. Also, I would double check that none of those fields can be `nil`. You're going quite deep with some (e.g. `cfg.LimitedHTTPClientSettings.TLSSetting.InsecureSkipVerify`)

Reply:
This is consistent with that we do for metrics exporter.  All the values `cfg.LimitedHTTPClientSettings.TLSSetting` are set for default config. 
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7f9c4d7ce8e435547f34c26d398c878e6cd3119f/exporter/datadogexporter/metrics_exporter.go#L103-L106

Reply:
may be we can do this as tidying up